src/conditions/condition-expr.ts

```diff
@@ -0,0 +1,108 @@
+import { Conditions as WASMConditions } from '@nucypher/nucypher-core';
+import { ethers } from 'ethers';
+
+import { objectEquals, toJSON } from '../utils';
+
+import {
+  Condition,
+  ContractCondition,
+  RpcCondition,
+  TimeCondition,
+} from './base';
+import { BLOCKTIME_METHOD } from './base/time';
+import { CompoundCondition } from './compound-condition';
+import { ConditionContext } from './context';
+
+export type ConditionExpressionJSON = {
+  version: string;
+  condition: Record<string, unknown>;
+};
```

Comment:
Does that limit the allowed properties as a whole, or ensures that at least one of those property values is present?

Reply:
It only defines a set of legal keys in `Record`. So if the key is present, it has to be of the given type, but if there are no keys it's still considered a valid object. I.e. `{}` is a valid `condition` field.

It's essentially adding type-checking, it does nothing for run-time validation.

Reply:
I think you're right, I didn't consider other keys. So the `ConditionProps` should be defined as follows:

```
type ConditionProps = 'operator' | 'method' | 'contractAddress' | string; // or replacing string with all other possible keys
```

If we use `string`, I don't think this addition would be useful enough to consider, since `'operator' | 'method' | 'contractAddress' | string;` simplifies to just `string`.

`Record<string, unknown>` shows up in a couple of different places in the codebase, so it may be worthwhile (from the perspective of idiomatic TypeScript, etc.) to turn it into a custom `type`. But perhaps this is something we could look into another time.

Reply:
I think ultimately then we would be repeating the work of the underlying condition JOI schema. 

Perhaps we can use the JOI schema fields defined by the various underlying conditions instead of listing them out ourselves. But that can be considered in a separate issue/PR.

Reply:
Filed https://github.com/nucypher/nucypher-ts/issues/230.